eng/pipelines/templates/stages/platform-matrix.json

```diff
@@ -6,11 +6,13 @@
         {
             "OSConfiguration": {
                 "macOS-10.15": {
-                    "OSVmImage": "macOS-10.15"
```

Comment:
I think it would be better to add a build step/script that can calculate these values based on the pre-existing variables and agent characteristics, rather than embedding them into the matrix. The matrix should set values necessary for testing and it seems overcomplicated to bundle metric variables within it as well.

Reply:
I think this is a smarter approach. The one I had written did have quite the smell.